### PR TITLE
DOC: Improve docs of `jnp.var` and `std`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
 def var(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
         out: None = None, ddof: int = 0, keepdims: bool = False, *,
         where: ArrayLike | None = None, correction: int | float | None = None) -> Array:
+  r"""Compute the variance along a given axis.
+
+  JAX implementation of :func:`numpy.var`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the variance to be computed.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:797]: `axis` can't be an array
```suggestion
    axis: optional int or sequence of ints, default=None. Axis along which the variance to be computed.
```

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+      If None, variance is computed along all the axes.
+    dtype: The type of the output array. Default=None.
+    ddof: int, default=0. Degrees of freedom. The divisor in variance computation is
+      ``N-ddof``, ``N`` is number of elements along given axis.
+    keepdims: bool, default=False. If true, reduced axes are left in the result
+      with size 1.
+    where: int or array, default=None. The elements to be used in the variance.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:804]: `where` can't be an int
```suggestion
    where: optional boolean array, default=None. The elements to be used in the variance.
```

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+      Array should be broadcast compatible to the input.
+    correction: int or float, default=None. Array compatible name for ``ddof``.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:806]: I don't think "Array compatible name" is clear. Maybe say "alternative name for ddof"

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+      Both ddof and correction can't be provided simultaneously.
+    out: Unused by JAX.
+
+  Returns:
+    An array of the vairnace along the given axis.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:811]: variance

File: jax/_src/numpy/reductions.py [hunk 5]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+
+  See also:
+    - :func:`jax.numpy.mean`: Compute the mean of array elements over a given axis.
+    - :func:`jax.numpy.std`: Compute the standard deviation of array elements over
+      given axis.
+    - :func:`jax.numpy.nanvar`: Compute the variance along a given axis by ignoring
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:817]:
```suggestion
    - :func:`jax.numpy.nanvar`: Compute the variance along a given axis, ignoring
```

File: jax/_src/numpy/reductions.py [hunk 6]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+      NaNs values.
+    - :func:`jax.numpy.nanstd`: Computed the standard deviation of a given axis by
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:819]:
```suggestion
    - :func:`jax.numpy.nanstd`: Computed the standard deviation of a given axis,
```

File: jax/_src/numpy/reductions.py [hunk 7]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+      ignoring NaN values.
+
+  Examples:
+    By default, ``jnp.var`` computes the variance along all axes.
+
+    >>> x = jnp.array([[1, 3, 4, 2],
+    ...                [5, 2, 6, 3],
+    ...                [8, 4, 2, 9]])
+    >>> jnp.var(x)
+    Array(5.7430553, dtype=float32)
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:829]: This may cause docstest problems due to floating point precision issues.

File: jax/_src/numpy/reductions.py [hunk 8]
```diff
@@ -785,10 +785,81 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+
+    If ``axis=1``, variance is computed along axis 1.
+
+    >>> jnp.var(x, axis=1)
+    Array([1.25  , 2.5   , 8.1875], dtype=float32)
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:834]: These are probably fine in terms of doctest & float precision because the fractional parts are powers of 2, which can be represented exactly.

File: jax/_src/numpy/reductions.py [hunk 9]
```diff
@@ -850,10 +921,76 @@ def _var_promote_types(a_dtype: DTypeLike, dtype: DTypeLike | None) -> tuple[DTy
...
 def std(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None,
         out: None = None, ddof: int = 0, keepdims: bool = False, *,
         where: ArrayLike | None = None, correction: int | float | None = None) -> Array:
+  r"""Compute the standard deviation along a given axis.
+
+  JAX implementation of :func:`numpy.std`.
+
+  Args:
+    a: Input array.
+    axis: int or array, default=None. Axis along which the standard deviation to
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:933]: int or sequence of ints

File: jax/_src/numpy/reductions.py [hunk 10]
```diff
@@ -850,10 +921,76 @@ def _var_promote_types(a_dtype: DTypeLike, dtype: DTypeLike | None) -> tuple[DTy
...
+      be computed. If None, standard deviaiton is computed along all the axes.
+    dtype: The type of the output array. Default=None.
+    ddof: int, default=0. Degrees of freedom. The divisor in standard eviation
+      computation is ``N-ddof``, ``N`` is number of elements along given axis.
+    keepdims: bool, default=False. If true, reduced axes are left in the result
+      with size 1.
+    where: int or array, default=None. The elements to be used in the standard
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:940]: optional boolean array

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 6 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 7 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Changed the precision of print using `jnp.printoptions`.

File: jax/_src/numpy/reductions.py (hunk 9 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 10 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Implemented the suggested changes and squashed into single commit.

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Changed

File: jax/_src/numpy/reductions.py [hunk 11]
```diff
@@ -785,10 +785,82 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
         out: None = None, ddof: int = 0, keepdims: bool = False, *,
         where: ArrayLike | None = None, correction: int | float | None = None) -> Array:
+  r"""Compute the variance along a given axis.
+
+  JAX implementation of :func:`numpy.var`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      variance to be computed. If None, variance is computed along all the axes.
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:798]:
```suggestion
      variance is computed. If None, variance is computed along all the axes.
```

File: jax/_src/numpy/reductions.py [hunk 12]
```diff
@@ -850,10 +922,78 @@ def _var_promote_types(a_dtype: DTypeLike, dtype: DTypeLike | None) -> tuple[DTy
...
         out: None = None, ddof: int = 0, keepdims: bool = False, *,
         where: ArrayLike | None = None, correction: int | float | None = None) -> Array:
+  r"""Compute the standard deviation along a given axis.
+
+  JAX implementation of :func:`numpy.std`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      standard deviation to be computed. If None, standard deviaiton is computed
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:935]:
```suggestion
      standard deviation is computed. If None, standard deviation is computed
```

File: jax/_src/numpy/reductions.py [hunk 13]
```diff
@@ -850,10 +922,78 @@ def _var_promote_types(a_dtype: DTypeLike, dtype: DTypeLike | None) -> tuple[DTy
...
+      along all the axes.
+    dtype: The type of the output array. Default=None.
+    ddof: int, default=0. Degrees of freedom. The divisor in standard eviation
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:938]:
```suggestion
    ddof: int, default=0. Degrees of freedom. The divisor in the standard deviation
```

File: jax/_src/numpy/reductions.py [hunk 14]
```diff
@@ -785,10 +785,82 @@ def _average(a: ArrayLike, axis: Axis = None, weights: ArrayLike | None = None,
...
+  r"""Compute the variance along a given axis.
+
+  JAX implementation of :func:`numpy.var`.
+
+  Args:
+    a: input array.
+    axis: optional, int or sequence of ints, default=None. Axis along which the
+      variance to be computed. If None, variance is computed along all the axes.
+    dtype: The type of the output array. Default=None.
+    ddof: int, default=0. Degrees of freedom. The divisor in variance computation
```
Comment (dfm) [inline jax/_src/numpy/reductions.py:800]:
```suggestion
    ddof: int, default=0. Degrees of freedom. The divisor in the variance computation
```

**Review by dfm — CHANGES_REQUESTED** [review]
Thanks - looks good! I added some small edits inline.

File: jax/_src/numpy/reductions.py (hunk 11 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 12 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 13 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 14 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

**Comment by rajasekharporeddy** [comment]
Squashed into single commit.

**Review by dfm — APPROVED** [review]
Thanks!
